<a href="https://colab.research.google.com/github/Julio-M39/PesquisaAltosEnem/blob/master/PesquisaAltosEnem.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **PESQUISA DE DESEMPENHO DOS INSCRITOS NO ENEM 2016 a 2019 QUE MORAM OU MORAVAM NO MUNICÍPIO DE ALTOS NO ESTADO DO PIAUÍ**

## **1.0 Bibliotecas**

In [1]:
#Importando Bibliotecas
import pandas as pd

## **2.0 Lendo os Dados**

In [30]:
#Selecionando as bases de dados
fonte2016 = 'https://github.com/Julio-M39/PesquisaAltosEnem/blob/master/microdadosEnem2016.csv?raw=true'
fonte2017 = 'https://github.com/Julio-M39/PesquisaAltosEnem/blob/master/microdadosEnem2017.csv?raw=true'
fonte2018 = 'https://github.com/Julio-M39/PesquisaAltosEnem/blob/master/microdadosEnem2018.csv?raw=true'
fonte2019 = 'https://github.com/Julio-M39/PesquisaAltosEnem/blob/master/microdadosEnem2019.csv?raw=true'

In [31]:
#Leitura de dados
microdadosEnem2016 = pd.read_csv(fonte2016)
microdadosEnem2017 = pd.read_csv(fonte2017)
microdadosEnem2018 = pd.read_csv(fonte2018)
microdadosEnem2019 = pd.read_csv(fonte2019)

## **3.0 Explorando os Dados**

### **3.1 Visualização do dataframe**

In [32]:
#Visualização do dataframe
microdadosEnem2019.head(3)

Unnamed: 0  NU_INSCRICAO  NU_ANO  ... NU_NOTA_LC NU_NOTA_MT  NU_NOTA_REDACAO
0       91345  190001095974    2019  ...      562.8      516.0            640.0
1       91355  190001095984    2019  ...      539.0      462.5            560.0
2       91397  190001096026    2019  ...      455.3      413.3            720.0

[3 rows x 17 columns]

Visualização de apenas um dataframe pois os outros seguem o mesmo padrão de colunas, ou seja as colunas de todos os dataframes são iguais

### **3.2 Números de inscritos Altoenses por ano**

In [33]:
#Verificando se existem inscrições em branco
print(microdadosEnem2019['NU_INSCRICAO'].isnull().sum())


0


Todas as linhas de inscritos estão preenchidas

In [34]:
#Número de inscritos
print(microdadosEnem2019['NU_INSCRICAO'].shape,
      microdadosEnem2018['NU_INSCRICAO'].shape,
      microdadosEnem2017['NU_INSCRICAO'].shape,
      microdadosEnem2016['NU_INSCRICAO'].shape)

(1507,) (1439,) (1979,) (2450,)


Números de inscritos que residem em Altos por ano:

* Ano de 2019 - 1507 Inscritos
* Ano de 2018 - 1439 Inscritos
* Ano de 2017 - 1979 Inscritos
* Ano de 2016 - 2450 Inscritos

### **3.3 Números de inscritos Altoenses pelo município da escola onde estudaram**

#### **2019**

In [35]:
#Verificar quantos inscritos não responderam com o nome do município da escola em 2019
print(microdadosEnem2019['NO_MUNICIPIO_ESC'].isnull().sum())

1217


In [36]:
#Verificar os nomes dos municípios onde os inscritos Altoenses estudaram em 2019
microdadosEnem2019['NO_MUNICIPIO_ESC'].value_counts()

Altos                  233
Teresina                47
Campo Maior              7
Pau D'Arco do Piauí      2
Bom Jesus                1
Name: NO_MUNICIPIO_ESC, dtype: int64

Em 2019 os municípios onde os Altoenses estudaram foram:

* 233 Inscritos - Município de Altos
* 47 Inscritos - Município de Teresina 
* 7 Inscritos - Município de Campo Maior
* 2 Inscritos - Município de Pau D'Arco do Piauí
* 1 Inscritos - Município de Bom Jesus

1217 Inscritos não responderam com o nome do município da escola onde estudou ou estuda, totalizando os 1507 inscritos no ano de 2019.





#### **2018**

In [37]:
#Verificar quantos inscritos não responderam com o nome do município da escola em 2018
print(microdadosEnem2018['NO_MUNICIPIO_ESC'].isnull().sum())

1132


In [38]:
#Verificar os nomes dos municípios onde os inscritos Altoenses estudaram em 2018
microdadosEnem2018['NO_MUNICIPIO_ESC'].value_counts()

Altos                  257
Teresina                44
Pau D'Arco do Piauí      3
Campo Maior              2
São João da Serra        1
Name: NO_MUNICIPIO_ESC, dtype: int64

Em 2018 os municípios onde os Altoenses estudaram foram:

* 257 Inscritos - Município de Altos
* 44 Inscritos - Município de Teresina
* 3 Inscritos - Município de Pau D'Arco do Piauí
* 2 Inscritos - Município de Campo Maior
* 1 Inscritos - Município de São João da Serra

1132 Inscritos não responderam com o nome do município da escola onde estudou ou estuda, totalizando os 1439 inscritos no ano de 2018.

#### **2017**

In [39]:
#Verificar quantos inscritos não responderam com o nome do município da escola em 2017
print(microdadosEnem2017['NO_MUNICIPIO_ESC'].isnull().sum())

1564


In [40]:
#Verificar os nomes dos municípios onde os inscritos Altoenses estudaram em 2017
microdadosEnem2017['NO_MUNICIPIO_ESC'].value_counts()

Altos                  372
Teresina                34
Campo Maior              4
Pau D'Arco do Piauí      4
Coivaras                 1
Name: NO_MUNICIPIO_ESC, dtype: int64

Em 2017 os municípios onde os Altoenses estudaram foram:

* 372 Inscritos - Município de Altos
* 34 Inscritos - Município de Teresina
* 4 Inscritos - Município de Pau D'Arco do Piauí
* 4 Inscritos - Município de Campo Maior
* 1 Inscritos - Município de Coivaras

1564 Inscritos não responderam com o nome do município da escola onde estudou ou estuda, totalizando os 1979 inscritos no ano de 2017.

#### **2016**

In [41]:
#Verificar quantos inscritos não responderam com o nome do município da escola em 2016
print(microdadosEnem2016['NO_MUNICIPIO_ESC'].isnull().sum())

1983


In [42]:
#Verificar os nomes dos municípios onde os inscritos Altoenses estudaram em 2016
microdadosEnem2016['NO_MUNICIPIO_ESC'].value_counts()

Altos                  409
Teresina                42
Pau D'Arco do Piauí     13
Beneditinos              1
Corrente                 1
Santa Inês               1
Name: NO_MUNICIPIO_ESC, dtype: int64

Em 2016 os municípios onde os Altoenses estudaram foram:

* 409 Inscritos - Município de Altos
* 42 Inscritos - Município de Teresina
* 13 Inscritos - Município de Pau D'Arco do Piauí
* 1 Inscrito - Município de Corrente
* 1 Inscritos - Município de Santa Inês
* 1 Inscritos - Município de Beneditinos

1983 Inscritos não responderam com o nome do município da escola onde estudou ou estuda, totalizando os 2450 inscritos no ano de 2017.

### **3.4 Números de inscritos Altoenses pelo tipo de escola**

#### **3.4.1 Substituindo os valores da coluna TP_ESCOLA**

No dataframe a coluna TP_ESCOLA contêm valores de 1 a 4 para representar os tipos das escolas, sendo eles:

* 1 - Não Respondeu
* 2 - Pública
* 3 - Privada
* 4 - Exterior


In [43]:
#Substituindo os valores
microdadosEnem2019['TP_ESCOLA'] = microdadosEnem2019['TP_ESCOLA'].map({1: 'Não Respondeu' , 2: 'Pública', 3: 'Privada', 4: 'Exterior'})
microdadosEnem2018['TP_ESCOLA'] = microdadosEnem2018['TP_ESCOLA'].map({1: 'Não Respondeu' , 2: 'Pública', 3: 'Privada', 4: 'Exterior'})
microdadosEnem2017['TP_ESCOLA'] = microdadosEnem2017['TP_ESCOLA'].map({1: 'Não Respondeu' , 2: 'Pública', 3: 'Privada', 4: 'Exterior'})
microdadosEnem2016['TP_ESCOLA'] = microdadosEnem2016['TP_ESCOLA'].map({1: 'Não Respondeu' , 2: 'Pública', 3: 'Privada', 4: 'Exterior'})


In [44]:
#Novos valores
microdadosEnem2019['TP_ESCOLA'].value_counts()

Não Respondeu    1160
Pública           325
Privada            22
Name: TP_ESCOLA, dtype: int64

####**3.4.2 Números de inscritos Altoenses pelo tipo de escola por ano:**

##### **2019**

In [46]:
#Números de inscritos pelo tipo de escola
microdadosEnem2019['TP_ESCOLA'].value_counts()

Não Respondeu    1160
Pública           325
Privada            22
Name: TP_ESCOLA, dtype: int64

No ano de 2019 os inscritos Altoenses estudaram em escolas do tipo:

* 325 Inscritos - Colégio Público
* 22 Inscritos - Colégio Privado

1160 inscritos não respondeu em que tipo de escola estudaram totalizando os 1507 Altoenses inscritos no ano de 2019.

##### **2018**

In [47]:
#Números de inscritos pelo tipo de escola
microdadosEnem2018['TP_ESCOLA'].value_counts()

Não Respondeu    1145
Pública           266
Exterior           26
Privada             2
Name: TP_ESCOLA, dtype: int64

No ano de 2018 os inscritos Altoenses estudaram em escolas do tipo:

* 266 Inscritos - Colégio Público
* 26 Inscritos - Colégio Exterior
* 2 Inscritos - Colégio Privado

1145 inscritos não respondeu em que tipo de escola estudaram totalizando os 1439 Altoenses inscritos no ano de 2018.

##### **2017**

In [48]:
#Números de inscritos pelo tipo de escola
microdadosEnem2017['TP_ESCOLA'].value_counts()

Não Respondeu    1564
Pública           383
Privada            32
Name: TP_ESCOLA, dtype: int64

No ano de 2017 os inscritos Altoenses estudaram em escolas do tipo:

* 383 Inscritos - Colégio Público
* 32 Inscritos - Colégio Privado

1564 inscritos não respondeu em que tipo de escola estudaram totalizando os 1979 Altoenses inscritos no ano de 2018.

##### **2016**

In [49]:
#Números de inscritos pelo tipo de escola
microdadosEnem2016['TP_ESCOLA'].value_counts()

Não Respondeu    1983
Pública           424
Privada            43
Name: TP_ESCOLA, dtype: int64

No ano de 2016 os inscritos Altoenses estudaram em escolas do tipo:

* 424 Inscritos - Colégio Público
* 43 Inscritos - Colégio Privado

1983 inscritos não respondeu em que tipo de escola estudaram totalizando os 2450 Altoenses inscritos no ano de 2016.

## **4.0 Divisões dos datasets**